crates/bevy_ecs/src/component.rs

```diff
@@ -1702,22 +1725,81 @@ pub struct RequiredComponent {
+    pub(crate) required_components: HashMap<ComponentId, RelatedComponent>,
+}
 
-impl Debug for RequiredComponents {
+impl Debug for RelatedComponents {
     fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
-        f.debug_tuple("RequiredComponents")
-            .field(&self.0.keys())
+        f.debug_struct("RelatedComponents")
+            .field("Suggested", &self.suggested_components.keys())
+            .field("Included", &self.included_components.keys())
+            .field("Required", &self.required_components.keys())
             .finish()
     }
 }
 
-impl RequiredComponents {
-    /// Registers a required component.
+/// The strictness of the relationship between components.
+///
+/// These enum variants are ordered from least strict to most strict.
+///
+/// Related components are unidirectional: if `A` requires `B`, `B` does not necessarily require `A`.
+/// These variants describe the strength of the relationship from `A` to `B`, where `A` is the component
+/// that you are registering related components for.
+///
+/// In many cases, you might create an included/required component relationship in one direction,
+/// but merely suggest the relationship in the other direction to help users discover the connection.
+///
+/// For usage information, see the "Related Components" section of [`Component`].
+/// For the collection of metadata associated with a specific component, see [`RelatedComponents`].
+/// For a single piece of metadata associated with a related component, see [`RelatedComponent`].
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+pub enum Relatedness {
+    /// This component has a helpful interaction with the related component.
+    ///
+    /// At this level, documentation is automatically generated (when used as part of the derive macro),
+    /// and runtime metadata linking the components is stored for use with tooling.
+    ///
+    /// This level is appropriate to link components that have a helpful, optional effect when used together.
+    Suggested,
+    /// This component should typically be used with the related component.
+    ///
+    /// In addition to the features of [`Relatedness::Suggested`],
+    /// whenever this component is added to an entity, the related component will also be added if it was not already provided.
+    ///
+    /// This level is appropriate to link components that should almost always be used together,
+    /// but where power users may wish to opt-out of this behavior.
+    Included,
+    /// This component must be used with the related component.
+    ///
+    /// In addition to the features of [`Relatedness::Suggested`],
+    /// whenever this component is added to an entity, the relationship between these components cannot be removed.
+    /// In the future, we intend to this level to enforce even stricter guarantees, such as removing this component
+    /// if any of its required component is removed.
```

Comment:
I think this section maybe got interrupted part way through editing after duplicating from the `Included` comment, perhaps:

```suggestion
    /// In addition to the features of [`Relatedness::Suggested`],
    /// whenever this component is added to an entity, the related component will also be added if it was not already provided.
    /// The relationship between these components cannot be removed.
    /// In the future, we intend to this level to enforce even stricter guarantees, such as removing this component
    /// if any of its required component is removed.
```